openshift/template.yaml

```diff
@@ -5,304 +5,216 @@ labels:
               secretKeyRef:
-                name: ceph-secrets
+                name: thoth
                 key: ceph-key-id
           - name: THOTH_CEPH_SECRET_KEY
             valueFrom:
               secretKeyRef:
-                name: ceph-secrets
+                name: thoth
                 key: ceph-secret-key
 
-- apiVersion: v1
-  kind: Service
-  metadata:
-    labels:
-      app: graphexp
-      service: graphexp
-    name: graphexp
-    namespace: ${THOTH_FRONTEND_NAMESPACE}
-  spec:
-    ports:
-    - port: 80
-      protocol: TCP
-      targetPort: 8080
-    selector:
-      service: graphexp
-
 #
 # Routes
-#
-- apiVersion: v1
-  kind: Route
+- kind: Route
+  apiVersion: v1
   metadata:
-    name: thoth-user-api-hostname
-    namespace: ${THOTH_FRONTEND_NAMESPACE}
+    name: user-api
   spec:
     host: ${THOTH_USER_API_HOSTNAME}
     to:
       kind: Service
-      name: thoth-user-api
+      name: user-api
 
-- apiVersion: v1
-  kind: Route
+- kind: Route
+  apiVersion: v1
   metadata:
-    name: graphexp-hostname
-    labels:
-      app: graphexp
-    namespace: ${THOTH_FRONTEND_NAMESPACE}
+    name: result-api
   spec:
-    host:
+    host: ${THOTH_RESULT_API_HOSTNAME}
     to:
       kind: Service
-      name: graphexp
-
-#
-# Service accounts
-#
-- apiVersion: v1
-  kind: RoleBinding
-  metadata:
-    name: role-jobs-binding
-    namespace: ${THOTH_MIDDLEEND_NAMESPACE}
-  roleRef:
-    kind: ClusterRole
-    name: edit
-  subjects:
-    - kind: ServiceAccount
-      name: master-access
-      namespace: ${THOTH_FRONTEND_NAMESPACE}
-
-- apiVersion: v1
-  kind: ServiceAccount
-  metadata:
-    name: master-access
-    namespace: ${THOTH_FRONTEND_NAMESPACE}
-  automountServiceAccountToken: true
-
-- apiVersion: v1
-  kind: RoleBinding
-  metadata:
-    name: role-jobs-binding
-    namespace: ${THOTH_BACKEND_NAMESPACE}
-  roleRef:
-    kind: ClusterRole
-    name: edit
-  subjects:
-    - kind: ServiceAccount
-      name: master-access
-      namespace: ${THOTH_FRONTEND_NAMESPACE}
-
+      name: result-api
 #
 # Deployment configs
 #
 - apiVersion: v1
   kind: DeploymentConfig
   metadata:
     labels:
-      service: thoth-user-api
-    name: thoth-user-api
-    namespace: ${THOTH_FRONTEND_NAMESPACE}
+      service: user-api
+    name: user-api
   spec:
     replicas: 1
     selector:
-      service: thoth-user-api
+      service: user-api
     template:
       metadata:
         labels:
-          service: thoth-user-api
+          service: user-api
       spec:
-        serviceAccountName: master-access
+        serviceAccountName: analyzer
         containers:
         - env:
           - name: APP_SERVICE_PORT
-            value: "8080"
+            valueFrom:
+              configMapKeyRef:
+                key: user-api-service-port
+                name: thoth
           - name: KUBERNETES_API_URL
-            value: ${KUBERNETES_API_URL}
+            value: 'https://kubernetes.default.svc.cluster.local'
```

Comment:
I would like to keep this configurable. If we will get some kubernetes cluster (e.g. hosted on GKE) that we would like to orchestrate from user API service, we can do so by adjusting configuration without a need to deploy whole thoth-core. It will be just a pool of resources we could use for middleend/backaned. The same applies to result API.

Reply:
But the cluster internal API URL of Kubernetes is generic, why should we configure it?

Reply:
If we will have a hosted cluster somewhere with some specific hardware, we don't need to deploy the whole thoth there, but we can configure api service to submit analyzes to this cluster by providing kubernetes API and kubernetes token - the same way how oc binary communicates with openshift.

This is internal cluster url, but it can be also external where we would also supply API token to communicate with kubernetes master [1] (defaults to service account).

[1] https://github.com/fridex/thoth-user-api/blob/master/thoth_user_api/configuration.py#L24

Reply:
hmm, understood. This feature doesnt seem to be on the critical path, I would like to focus on getting a CI/CD pipeline up and running

Reply:
This feature is already there - there is only required to adjust parameters of deployment if we would like to do that.